core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -455,7 +455,9 @@ private[chisel3] class DynamicContext(
   val defaultNamespace:      Option[Namespace],
   // Definitions from other scopes in the same elaboration, use allDefinitions below
   val outerScopeDefinitions: List[Iterable[Definition[_]]],
-  val loggerOptions:         LoggerOptions) {
+  val loggerOptions:         LoggerOptions,
+  val definitions:           ArrayBuffer[Definition[_]],
```

Comment:
See `outerScopeDefinitions` above (added in https://github.com/chipsalliance/chisel/pull/3895). I added that to fix passing Definitions down to other Definitions, but it does essentially remove nested definitions. I think just switching to use the `ArrayBuffer` is probably right (I'm unsure why I was thinking we need to remove definitions added by the nested Definitions...).

Regardless, please remove `outerScopeDefinitions` and make sure everything works with just definitions because this change makes `outerScopeDefinitions` very weird (everything in the list is going to be the same ArrayBuffer).

Reply:
Hahahahahaha, I thought about this, but dear not to touch it...
Will do it tonight.